### PR TITLE
Removing Binary serialization

### PR DESCRIPTION
**Pull request by joerivanarkel**
**Summary**:
- Replacing Binary serialization with JSON serialization
    - Changed interpreters to use `Int64` instead of `int` and `double` instead of `single`
    - Refined token to just use `TokenGroup` and `TokenType` enums
- Added Logo to `README.md`
- Refactored Parser and Interpreter methods
- Added XML documentation comment per class

### CODE REVIEW

File: PirateInterpreter/Interpreters/BaseInterpreter.cs [hunk 1]
```diff
@@ -18,6 +18,7 @@ public BaseInterpreter(ILogger logger, InterpreterFactory interpreterFactory)
     public BaseValue VisitSingleNode()
     {
```
Comment (teusvanarkel) [inline PirateInterpreter/Interpreters/BaseInterpreter.cs:19]: Check this method maybe can improved

File: PirateInterpreter/Interpreters/ComparisonOperationInterpreter.cs [hunk 2]
```diff
@@ -27,24 +27,24 @@ public override List<BaseValue> VisitNode()
 
         switch (operationNode.Operator.TokenType)
         {
-            case TokenComparisonOperators.DOUBLEEQUALS:
+            case TokenType.DOUBLEEQUALS:
```
Comment (teusvanarkel) [inline PirateInterpreter/Interpreters/ComparisonOperationInterpreter.cs:30]: Check all values for using Int64

File: PirateInterpreter/Values/BaseValue.cs [hunk 3]
```diff
@@ -20,7 +20,7 @@ public int Matches(BaseValue other)
     {
         if (Value.GetType() != other.Value.GetType())
         {
-            Logger.Log($"Types dont match. {Value} | {other.Value}", Common.Enum.LogType.INFO);
+            Logger.Log($"Types dont match. {Value.GetType()} : {Value} | {other.Value.GetType()} : {other.Value}", Common.Enum.LogType.INFO);
             return 0;
```
Comment (teusvanarkel) [inline PirateInterpreter/Values/BaseValue.cs:24]: Make bool and return true and false

File: PirateLexer/Tokens/KeyWordService.cs [hunk 4]
```diff
@@ -9,71 +9,71 @@ public class KeyWordService : IKeyWordService
 
     private string[] controlKeywords = new string[] { "if", "else", "for", "to", "foreach", "in", "while", "func", "class", "new", "return" };
 
-    public TokenTypeKeyword GetTypeKeyword(string idString)
+    public TokenType GetTypeKeyword(string idString)
```
Comment (teusvanarkel) [inline PirateLexer/Tokens/KeyWordService.cs:12]: Change to C11 switch expression

Comment (teusvanarkel) [reply PirateLexer/Tokens/KeyWordService.cs:12]: example:
    public static Orientation ToOrientation(Direction direction) => direction switch
    {
        Direction.Up    => Orientation.North,
        Direction.Right => Orientation.East,
        Direction.Down  => Orientation.South,
        Direction.Left  => Orientation.West,
        _ => throw new ArgumentOutOfRangeException(nameof(direction), $"Not expected direction value: {direction}"),
    };

File: PirateInterpreter/Values/BaseValue.cs (hunk 3 above)
Comment (joerivanarkel) [reply PirateInterpreter/Values/BaseValue.cs:23]: has a specific reason to return int

**Pull request merged**